src/test/java/com/onfido/integration/ReportSchemasTest.java

```diff
@@ -26,18 +26,18 @@ public void schemaOfDocumentReportIsValid() throws Exception {
             document,
             new CheckBuilder().reportNames(Arrays.asList(ReportName.DOCUMENT)));
 
-    DocumentReport documentReport =
-        (DocumentReport)
+    Report documentReport =
+        (Report)
             repeatRequestUntilStatusChanges(
-                "findReport", new Object[] {check.getReportIds().get(0)}, COMPLETE, 10, 1000);
+                "findReport", new Object[] {check.getReportIds().get(0)}, COMPLETE, 15, 1000);
 
-    Assertions.assertEquals(COMPLETE, documentReport.getStatus());
+    Assertions.assertEquals(COMPLETE, documentReport.getReportShared().getStatus());
     Assertions.assertEquals(ReportName.DOCUMENT, documentReport.getName());
-    Assertions.assertEquals(check.getId(), documentReport.getCheckId());
-    Assertions.assertNotNull(documentReport.getBreakdown());
-    Assertions.assertNotNull(documentReport.getProperties());
-    Assertions.assertNotNull(documentReport.getResult());
-    Assertions.assertNotNull(documentReport.getSubResult());
+    Assertions.assertEquals(check.getId(), documentReport.getReportShared().getCheckId());
+    Assertions.assertNotNull(((DocumentReport) documentReport.getActualInstance()).getBreakdown());
```

Comment:
There is no need to call `getActualInstance()` and cast the result. Just calling `getDocumentReport()` (or `getFacialSimilarityPhotoFullyAutoReport()` later on) would do the trick. ;)